packages/aws-cdk-lib/aws-lambda/lib/code.ts

```diff
@@ -54,6 +55,44 @@ export abstract class Code {
     return new AssetCode(path, options);
   }
 
+  /**
+   * Runs a command to build the code asset that will be used.
+   *
+   * @param output Where the output of the command will be directed, either a directory or a .zip file with the output Lambda code bundle
+   * * For example, if you use the command to run a build script (e.g., [ 'node', 'bundle_code.js' ]), and the build script generates a directory `/my/lambda/code`
+   * containing code that should be ran in a Lambda function, then output should be set to `/my/lambda/code`
+   * @param command The command which will be executed to generate the output, for example, [ 'node', 'bundle_code.js' ]
+   * @param assetOptions The same options that are available for `Code.fromAsset` -- but bundling options are not allowed
+   * @param commandOptions Options that are passed to the spawned process, which determine the characteristics of the spawned process.
+   * * See `child_process.SpawnSyncOptions` for possible inputs and defaults (https://nodejs.org/api/child_process.html#child_processspawnsynccommand-args-options).
+   */
+  public static fromCustomCommand(
+    output: string,
+    command: string[],
+    commandOptions?: {[option: string]: any}, // jsii build fails if the type is SpawnSyncOptions... so best we can do is point user to those options.
+    assetOptions?: s3_assets.AssetOptions,
```

Comment:
I generally prefer to use interfaces where we can so that if there is ever a chance that a new argument is introduced we can just add it to the interface. What do you think about creating a new interface that just extends `AssetOptions`? We could call it `CustomCommandOptions`. Something like this:

```ts
export interface CustomCommandOptions extends AssetOptions {
	readonly commandOptions?: { [options: string]: any },
}
```

Then the function definition could just be:

```ts
public static fromCustomCommand(output: string, command: string[], options: CustomCommandOptions = {})
```

Reply:
I agree with you, that's an improvement